pocs/utils/rs232.py

```diff
@@ -12,56 +12,66 @@
 
 
 class SerialData(PanBase):
-
     """
     Main serial class
     """
 
-    def __init__(self, port=None, baudrate=115200, threaded=True, name="serial_data"):
+    def __init__(self,
+                 port=None,
+                 baudrate=115200,
+                 threaded=True,
+                 name="serial_data",
+                 open_delay=2.0):
         PanBase.__init__(self)
 
-        try:
-            self.ser = serial.Serial()
-            self.ser.port = port
-            self.ser.baudrate = baudrate
-            self.is_threaded = threaded
-
-            self.ser.bytesize = serial.EIGHTBITS
-            self.ser.parity = serial.PARITY_NONE
-            self.ser.stopbits = serial.STOPBITS_ONE
-            self.ser.timeout = 1.0
-            self.ser.xonxoff = False
-            self.ser.rtscts = False
-            self.ser.dsrdtr = False
-            self.ser.write_timeout = False
-            self.ser.open()
-
-            self.name = name
-            self.queue = deque([], 1)
-            self._is_listening = False
-            self.loop_delay = 2.
-
-            if self.is_threaded:
-                self._serial_io = TextIOWrapper(BufferedRWPair(self.ser, self.ser),
-                                                newline='\r\n', encoding='ascii', line_buffering=True)
+        self.ser = serial.serial_for_url(port, do_not_open=True)
+        self.ser.baudrate = baudrate
+        self.is_threaded = threaded
+
+        self.ser.bytesize = serial.EIGHTBITS
+        self.ser.parity = serial.PARITY_NONE
+        self.ser.stopbits = serial.STOPBITS_ONE
+        self.ser.timeout = 1.0
+        self.ser.xonxoff = False
+        self.ser.rtscts = False
+        self.ser.dsrdtr = False
+        self.ser.write_timeout = False
+
+        self.name = name
+        self.queue = deque([], 1)
+        self._is_listening = False
+        self.process = None
+        self._serial_io = None
+        self.loop_delay = 2.
 
-                self.logger.debug("Using threads (multiprocessing)")
-                self.process = Thread(target=self.receiving_function, args=(self.queue,))
-                self.process.daemon = True
-                self.process.name = "PANOPTES_{}".format(name)
+        # Properties have been set to reasonable values, ready to open the port.
+        self.ser.open()
```

Comment:
Since we don't have any "open on write" behavior, this means that we have a completely useless SerialData instance if we ignore the failure. And since we use exceptions elsewhere for failures, I didn't see the point in hiding this failure. All the tests passed after making that change.